### DISCUSSION TITLE
Halo2 Book 里的 cryptography 问题

### DISCUSSION THREAD

**Discussion by Demian101** [z2o-k7e/halo2-step-by-step#24] [Q&A]
在 ZCash 的 Halo2 Book 中, 在 [ECC](https://zcash.github.io/halo2/design/gadgets/ecc.html) 这章，看到如下表述：

<img width="560" alt="image" src="https://github.com/zkp-co-learning/halo2-step-by-step/assets/33189338/63a0e666-e014-44c8-8a41-78cb6d1a3165">

没看懂是什么意思

chatgpt-4 对其解释地也不是很清楚
<img width="631" alt="image" src="https://github.com/zkp-co-learning/halo2-step-by-step/assets/33189338/d834ba05-1ce1-435d-a6b0-db5ab8ff3654">

BTW，作为一个密码学小白，我想知道类似在 Halo2 源码中的类似 `CurveAffine` `Pasta`  这些 密码学的知识，学得很懵，这些知识 Chatgpt 也一般回答地很差，应该如何学习，在写电路时是否应该（或者说如何）关注这些具体的密码学知识？

**Comment by Demian101** [comment]
<img width="808" alt="image" src="https://github.com/zkp-co-learning/halo2-step-by-step/assets/33189338/b4d10616-73f7-4551-a97b-cac6b0c1aa93">

and  没明白这个 degree 为啥是 4 🤣  不应该是 3 嘛

**Reply by qizhou** [reply]
q也算是一个未知数吗？
